answers/templates/queue.html

```diff
@@ -14,101 +12,120 @@ <h1>{{ hunt_name }} - Answer Queue</h1>
+        notes = 8;
+
+    let statuses = ['NEW', 'SUBMITTED', 'CORRECT', 'INCORRECT', 'PARTIAL',];
 
     $('#queue').DataTable( {
+        "ajax": "/answers/queue/{{hunt_pk}}/answers",
         "paging": true,
+        "pageLength": 50,
         "info": true,
         "scrollX": true,
         "scrollY": true,
+        "order": [[ time, "desc" ]],
+        "ordering": false,
+        "createdRow": function(row, data, dataIndex) {
+            $(row).addClass(data[answer_class]);
+        },
         "columnDefs": [
-            { "searchable": false, "targets": time },
             {
-              "targets": time,
-              "type": "date",
-              "render": function(data, type, row, meta) {
-                if (type === 'sort') {
-                    data = moment(data, 'MMM D, YYYY, h:m a');
-                }
-                return data;
-              },
+                'visible': false,
+                'targets': [answer_class, id, puzzle_url, puzzle_is_meta],
             },
-            { "orderable": false,
-              "targets": status,
-              "render": function(data, type, row, meta) {
-                if (type === 'filter') {
-                    data = $.parseHTML(data);
-                    data = data[0].querySelector('select').value;
-                }
-                return data;
-              },
+            {
+                'searchable': false,
+                'targets': [answer_class, id, time, puzzle_url, puzzle_is_meta],
             },
-            { "orderable": false,
-              "targets": notes,
-              "render": function(data, type, row, meta) {
-                if (type === 'filter') {
-                    data = $.parseHTML(data);
-                    data = data[0].querySelector('.form-control').value;
-                }
-                return data;
-              },
+            {'title': 'Time', 'targets': time},
+            {'title': 'Puzzle', 'targets': puzzle_name},
+            {'title': 'Answer', 'targets': guess},
+            {'title': 'Status', 'targets': answer_status},
+            {'title': 'Notes', 'targets': notes},
+
+            {
+                'targets': time,
+                'render': function(data, type, row, meta) {
+                    return moment(data).format('MMM D, YYYY, h:m a');
+                },
+            },
+            {
+                'targets': puzzle_name,
+                'render': function(data, type, row, meta) {
+                    let result = '<a href="' + row[puzzle_url] + '">' + row[puzzle_name] + '</a>';
+                    if (row[puzzle_is_meta]) {
+                        result += ' <span class="badge badge-dark badge-meta">META</span>';
+                    }
+                    return result;
+                },
+            },
+            {
+                'targets': guess,
+                'render': function(data, type, row, meta) {
+                    return `<p class="text-monospace">${data}</p>`
```

Comment:
Cool! ES6 template literals! I didn't realize we were using ES6; I guess it's pretty well-supported outside of IE11, and we don't care about that.

Reply:
Haha, yeah, I learned that from @rawxfish's PR here: https://github.com/cardinalitypuzzles/smallboard/pull/86/files#diff-a36dd12b9d7f908e5012c640b4326239R30

Also working on these changes, I learned about the `let` keyword (instead of `var`) and `for (x ... of y)` loops.

Reply:
It does seem like JavaScript is getting a little better

Reply:
I'm actually pretty impressed JavaScript now has these template literals that can be used for multi-line strings. Java doesn't even have such multi-line string literal support!